Login with mail or username

JethroWarnett opened the pull request:
If the user tries to login into his account, he/she can either send his e-mail address or his username, with a password.

A contract requirement, is that the frontend must send all information, even if the user just uses either email  or username

JethroWarnett committed:
Login with mail or username

If the user tries to login into his account, he/she can either send his e-mail address or his username, with a password.

A contract requirement, is that the frontend must send all information, even if the user just uses either email  or username

dominiqueroux commented on backend/src/services/user.service.ts:
```diff
@@ -15,11 +16,13 @@ export class UserService {
         })
         .then(user => {
-            if (bcrypt.compareSync(loginRequestee.password, user.password)) {// compares the hash with the password from the lognin request
+            if (bcrypt.compareSync(loginRequestee.password, user.password)) {// compares the hash with the password from the login request
```
😄

dominiqueroux commented on backend/src/services/user.service.ts:
```diff
@@ -15,11 +16,13 @@ export class UserService {
-                userName: loginRequestee.userName
+                [Op.or]: [
+                    {userName: loginRequestee.userName},
+                    {email: loginRequestee.email}]
```
We could use the following, which would make the life easier of the frontend team:
```suggestion
                    {userName: loginRequestee.userName},
                    {email: loginRequestee.userNamel}]
```
If we use the userName attribute from the loginRequestee model, the frontend wouldn't not need to provide both attritbutes

dominiqueroux commented on backend/src/models/login.model.ts:
```diff
@@ -8,5 +8,6 @@ export interface LoginResponse {
 
 export interface LoginRequest {
     userName: string;
+    email: string;
```
See comment below, probably this is not needed

JethroWarnett committed:
Update backend/src/services/user.service.ts

Co-authored-by: Dominique Roux <dominique.roux@protonmail.com>

dominiqueroux reviewed (commented):
I would not update the db.sqlite, since I assume you didn't change the DB scheme.

JethroWarnett committed:
Changed Interface of LoginResponse

Instead of having to provide both an e-mail and a username, the frontend can provide either in the attribute userNameOrEmail
